### PR TITLE
OGR/GDAL: quote/escape username and password when needed

### LINKED ISSUE #54493: QGIS cannot connect to MySQL if user's password contains commas.
### What is the bug or the crash?

When QGIS tries to connect to MySQL, the connection fails if the user's password contains commas.

### Steps to reproduce the issue

1. Create user at MySQL Server (I tried with MySQL 8.1.0)
2. Make password of the user to contain commas.
3. Create spatial table at MySQL.
4. Try to connect from QGIS to MySQL through Layer > Data Source Manager
5. Connection failed.

### Versions

QGIS version
3.30.3-'s-Hertogenbosch
QGIS code revision
2411cec1
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.7.0
PROJ version
9.2.0
EPSG Registry database version
v10.082 (2023-02-06)
GEOS version
3.11.2-CAPI-1.17.2
SQLite version
3.41.1
PDAL version
2.5.3
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

QGIS stores Connection string like
'MySQL:geo_test,host=localhost,port=3306,user=root,password=XXXXXXX'
So if password contains commas, password is splitted by comma.
<aybe this is the reason of this bug.

### PR DESCRIPTION
**Pull request by elpaso**
Fixes https://github.com/qgis/QGIS/issues/54493

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]
Can we get a test for this?

**Comment by elpaso** [comment]
> Can we get a test for this?

Yes, done!

File: src/auth/basic/core/qgsauthbasicmethod.cpp [hunk 1]
```diff
@@ -139,6 +139,21 @@ bool QgsAuthBasicMethod::updateDataSourceUriItems( QStringList &connectionItems,
   // Branch for OGR
   if ( dataprovider == QLatin1String( "ogr" ) || dataprovider == QLatin1String( "gdal" ) )
   {
+    // If username or password contains comma or double quote we need to quote the string
```
Comment (rouault) [inline src/auth/basic/core/qgsauthbasicmethod.cpp:142]: I'm afraid this logic won't work with all OGR drivers, and should probably be conditionalized to the MySQL one, given that code below for the PG driver uses single quotes

Comment (elpaso) [reply src/auth/basic/core/qgsauthbasicmethod.cpp:142]: Ok, but the question is:

Is it documented somewhere?

Comment (rouault) [reply src/auth/basic/core/qgsauthbasicmethod.cpp:142]: > Ok, but the question is:
> 
> Is it documented somewhere?

for PG at least: https://gdal.org/drivers/vector/pg.html#connecting-to-a-database

Comment (elpaso) [reply src/auth/basic/core/qgsauthbasicmethod.cpp]: Ok done, I suppose we should implement a similar logic for the other drivers but as far as this issue is concerned it is fixed.

**Pull request merged**